### ISSUE TITLE
Error MzidReader

### ISSUE THREAD

**Issue by Rebecca353** [compomics/psm_utils#30]
I experience an error when trying to open a specific mzid file.
https://ftp.pride.ebi.ac.uk/pride/data/archive/2023/02/PXD039808/ 
File: Brain_Regions_LFQ.mzid
<img width="1471" alt="Screenshot 2023-04-18 at 10 21 38" src="https://user-images.githubusercontent.com/131146749/232720095-09ccf126-bfe7-405a-952d-8f44e56a6a3e.png">

**Comment by RalfG** [comment]
Hi @Rebecca353,

Thanks for sharing the error! The issue originates from the fact that the `PeptideEvidence` entries are missing the `isDecoy` field.

From `Brain_Regions_LFQ.mzid`:
```
<PeptideEvidence post="A" pre="K" end="67" start="61" peptide_ref="PEP_-157_6596_88626" dBSequence_ref="-9221829543246918026" id="PE_88626_PEP_-157_6596_88626_-9221829543246918026" />
```

From an MS-GF+ mzid:
```
<PeptideEvidence dBSequence_ref="DBSeq258821" peptide_ref="Pep_LYDLGVQGADLIAMNTDAQHLAITK" start="52" end="76" pre="R" post="A" isDecoy="false" id="PepEv_258872_LYDLGVQGADLIAMNTDAQHLAITK_52"/>
```

I'm implementing a simple fix that will keep the internal `is_decoy` property for a PSM as `None` if the field is missing from the file. You could then potentially use the [PSMList.find_decoys](https://psm-utils.readthedocs.io/en/v0.2.3/api/psm_utils/#psm_utils.psm_list.PSMList.find_decoys) method to assign the property based on the protein names.

I will update this issue when the fix has been implemented in a new version of psm_utils.

Best,
Ralf

**Comment by RobbinBouwmeester** [comment]
Still get the error in (as the isDecoy key is not present):

`    @staticmethod
    def _parse_peptide_evidence_ref(peptide_evidence_list: list[dict]):
        """Parse peptide evidence References of PSM."""
        isdecoy = peptide_evidence_list[0]["isDecoy"]
        protein_list = [
            d["accession"] for d in peptide_evidence_list if "accession" in d.keys()
        ]
        return isdecoy, protein_list`

**Comment by RalfG** [comment]
Fixed in 69f6a46dd89ca7f78f4c8e634f08b30ce6ebb5a2. Seems like an older psm_utils version was used.
